zerver/openapi/js_examples.js

```diff
@@ -0,0 +1,68 @@
+const zulip = require('zulip-js');
+const fs = require('fs');
+
+const config = {
+    username: process.env.ZULIP_USERNAME,
+    apiKey: process.env.ZULIP_API_KEY,
+    realm: process.env.ZULIP_REALM,
+};
+
+// Object with an array of responses from the
+// code examples. It is written to 'js_example_responses.json'
+// later.
+const example_responses = {
+    responses: [],
+};
+
+const example_snippets = {};
+
+const examples = {};
+// Gets the responses from the code examples and
+// puts them in examples_output.responses .
+async function process_data() {
+    let response;
+    for (const method in examples) {
+        response = await examples[method](config, zulip);
+        const obj = {};
+        obj[method] = response;
+        example_responses.responses.push(obj);
+    }
+    return example_responses;
+}
+
+function generate_code_examples() {
+    for(let method in examples) {
+        example_snippets[method] = examples[method].toString();
+    }
+    fs.writeFileSync('./zerver/openapi/js_example_code.json', JSON.stringify(example_snippets, null, 4));
+}
+
+examples["/messages:post"] = async function send_message(config, zulip) {
+    // {code_example|start}
+    const client = await zulip(config);
+    const params = {
+        to: 'Denmark',
+        type: 'stream',
+        subject: 'Castle',
+        content: 'I come not, friends, to steal away your hearts.'
+    }
+
+    return await client.messages.send(params);
+    // {code_example|end}
+};
+
+const command = process.argv[2];
+switch (command) {
+case 'run-examples':
+    process_data().then(data => {
+        fs.writeFileSync('./zerver/openapi/js_example_responses.json', JSON.stringify(data, null, 4));
```

Comment:
We can do await with writing to a file. Just `console.log` the final JSON and have python subprocess read that information, parse it as JSON in python code.

Reply:
@aero31aero `send_message` demonstrates sending a stream message as well as a private message. Thus, it will generate two responses. If we `console.log` the final JSON, how will we validate multiple responses belonging to the same example.
In the other PR, we were running each example one by one and validating there responses in a loop.
How should we do this here ?

Reply:
I was thinking of writing them to the json file like:
```json
    {
        "/messages:post:200" : [{ .....response...}, {...response...}],
        "/users:post:200": [{...response...}]
    }
```

Reply:
We can validate in a the responses for each key in a loop then.